### PR TITLE
Shipping Labels Onboarding: first CTA screen

### LINKED ISSUE #6915: Shipping Labels Onboarding CTA screen
As part of the Shipping Labels Onboarding project, we should implement the first screen that is the entry point of the flow after tapping the banner in the order detail screen.

The CTA screen is dismissable, after pressing the `Not now` button, because the extension can still be installed from settings (or Order Detail for the M1).
The button `What is WooCommerce Shipping?` should point to this [URL](https://woocommerce.com/document/woocommerce-shipping-and-tax/).

### PR DESCRIPTION
**Pull request by pmusolino**
Closes #6915 

### Description
As part of the Shipping Labels Onboarding project, we should implement the first screen which is the entry point of the flow after tapping the banner in the order detail screen.
The CTA screen is dismissable, after pressing the `Not now` button.

### Testing instructions
1. Launch the app, and select a website without the WCShip extension installed (you can create a Jurassic ninja website, like a did). The feature flag `shippingLabelsOnboardingM1` should be enabled (enabled by default in debug mode). You can also add `return true` under `OrderDetailsDataSource` -> `shouldAllowWCShipInstallation` property.
2. Open the Order List.
3. Select an order.
4. You should be able to see the onboarding banner of WCShip after the products list.
5. By tapping on the banner, you should see the new CTA screen.
6. Tap the button "What is WooCommerce Shipping?" -> You should see the web view displaying what is WCShip.
7. Tap the button "Not now" -> The view should be dismissed.

### Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift [hunk 1]
```diff
@@ -981,6 +981,24 @@ extension UIImage {
...
+
+    /// Icon Circular Document (used in WCShip onboarding)
+    ///
+    static var iconCircularDocument: UIImage {
+        return UIImage(named: "icon-circular-document")!
+    }
+
+    /// Icon Circular Time (used in WCShip onboarding)
+    ///
+    static var iconCircularTime: UIImage {
```
Comment (jaclync) [inline WooCommerce/Classes/Extensions/UIImage+Woo.swift]: nit: do we want to follow the naming for other image variables like `*Image` or `*Icon`?

File: WooCommerce/WooCommerceTests/Extensions/IconsTests.swift [hunk 2]
```diff
@@ -648,4 +648,16 @@ final class IconsTests: XCTestCase {
...
+
+    func test_icon_circular_rate_discount() {
+        XCTAssertNotNil(UIImage.iconCircularRateDiscount)
+    }
+
+    func test_icon_circular_document() {
+        XCTAssertNotNil(UIImage.iconCircularDocument)
+    }
+
+    func test_icon_circular_time() {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Extensions/IconsTests.swift]: super nit: might be nice to follow other test cases like `*_is_not_nil` 🙂

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        super.init(rootView: WCShipCTAView())
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift]: nit: we probably don't need this empty override?

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    enum Constants {
+        static let topMargin: CGFloat = 64
+        static let horizontalMargin: CGFloat = 32
+        static let verticalSpacing: CGFloat = 16
+        static let whatIsWCShipURL: URL = URL(string: "https://woocommerce.com/document/woocommerce-shipping-and-tax/")!
+    }
+
+    enum Localization {
+        static let viewTitle = NSLocalizedString("Fullfill your orders with WooCommerce Shipping",
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift]: Typo here 🙂 
```suggestion
        static let viewTitle = NSLocalizedString("Fulfill your orders with WooCommerce Shipping",
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+}
+
+
+struct WCShipCTAView: View {
+    @ObservedObject private var iO = Inject.observer
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift:16]: I was trying out Injection, and verified that I saw these in the console:
```
💉 InjectionIII connected */WooCommerce.xcworkspace
💉 Watching files under */WooCommerce.xcworkspace
```
but I didn't see any updates on the screen from local changes 🤔 not sure if I missed something?

**Review by jaclync — APPROVED** [review]
LGTM! :shipit: 

One question about the eligibility for the WCShip onboarding UI:

https://github.com/woocommerce/woocommerce-ios/blob/0eae2a783db5f68577d613bf58b42cdda73e744f/WooCommerce/Classes/ViewModels/Order%20Details/OrderDetailsDataSource.swift#L103

any reasons we don't want to show the WCShip upsell to sites that support IPP?

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift (hunk 5 above)
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/WCShip Installation Process/WCShipCTAView.swift:16]: Just try to connect inside the app to the path of the entire folder of the project and not just to the xcworkspace. It should work 👍

**Comment by pmusolino** [comment]
Sorry, @jaclync  for all this delay, these days are 🔥 

Made all the changes suggested 🙌 

> One question about the eligibility for the WCShip onboarding UI:
> any reasons we don't want to show the WCShip upsell to sites that support IPP?

Yes, there is a reason behind it. Basically, the section for buying Shipping Labels will be visible if the order is not eligible for IPP,  and since from the last screen of this flow, it will be possible to buy a shipping label, we don't want to allow users to install the plugin, configure it, then doesn't allow them to buy a shipping label for that specific order. It's better to display this banner just in eligible orders, following the same logic of the section for buying a new shipping label 👍

**Comment by jaclync** [comment]
> Made all the changes suggested 🙌

Thanks for making the changes @pmusolino, LGTM!
There seems to be a failing test: `OrderDetailsDataSourceTests.test_WCShip_installation_section_is_visible_for_eligible_order()`

> > One question about the eligibility for the WCShip onboarding UI:
> > any reasons we don't want to show the WCShip upsell to sites that support IPP?
> 
> Yes, there is a reason behind it. Basically, the section for buying Shipping Labels will be visible if the order is not eligible for IPP, and since from the last screen of this flow, it will be possible to buy a shipping label, we don't want to allow users to install the plugin, configure it, then doesn't allow them to buy a shipping label for that specific order. It's better to display this banner just in eligible orders, following the same logic of the section for buying a new shipping label 👍

Is purchasing a shipping label at the end of the IPP flow a new feature? I don't remember this while working on IPP last month 🤔

**Comment by pmusolino** [comment]
> There seems to be a failing test: 
> OrderDetailsDataSourceTests.test_WCShip_installation_section_is_visible_for_eligible_order()

I merged `trunk` in this branch since the change was made in the PR that was targeting originally this one. 🙇 


> Is purchasing a shipping label at the end of the IPP flow a new feature? I don't remember this while working on IPP last month

Sorry, re-reading the message I wrote something confusing. I meant that when an order is eligible for IPP, it will not be eligible for shipping labels. In the purchase flow for the installation of the shipping labels plugin, in the last screen, it will be possible to buy the label for the specific order, but it is something we do not want to allow because if the order is eligible for IPP we must not allow to install WCShip and allow users to buy the shipping label.

**Comment by jaclync** [comment]
> Sorry, re-reading the message I wrote something confusing. I meant that when an order is eligible for IPP, it will not be eligible for shipping labels. In the purchase flow for the installation of the shipping labels plugin, in the last screen, it will be possible to buy the label for the specific order, but it is something we do not want to allow because if the order is eligible for IPP we must not allow to install WCShip and allow users to buy the shipping label.

@pmusolino Could you briefly explain a bit more on why an order eligible for IPP isn't allowed to install WCShip and then purchase a shipping label? 🤔 Is it because if an order is paid in person with IPP, the merchant doesn't have to ship the product? What about a pre-order, or the merchant uses a mix of IPP and online payment for their orders?

**Comment by pmusolino** [comment]
@jaclync basically we are using the same logic behind displaying the section for buying a Shipping Label. Here **pdfdoF-6f** and here **PdfdoF-V** you can find the answers. TLDR; _In other words, it is not possible to create a shipping label until payment for the order has been collected._ I think that one of the reasons discussed at the time was also about the number of elements displayed in the order detail view in the case of IPP.

> What about a pre-order, or the merchant uses a mix of IPP and online payment for their orders?

This is a good question, and honestly, I don't remember what was the discussion about it. I don't know if this case was taken into consideration. I'm going to ask the team that worked on IPP, maybe they remember the reason.

**Pull request merged**